extensions/servlet/src/main/java/com/stormpath/sdk/servlet/mvc/FormController.java

```
@@ -149,18 +149,28 @@ protected Form createForm(HttpServletRequest request, HttpServletResponse respon
     }
 
     protected List<Field> createFields(HttpServletRequest request, boolean retainPassword) {
-        List<Field> fields = new ArrayList<Field>();
+        List<Field> fields = new ArrayList<>();
 
         for (Field templateField : formFields) {
             Field clone = templateField.copy();
 
             if (clone.isEnabled()) {
                 String val = fieldValueResolver.getValue(request, clone.getName());
-                if (clone.getName() == "password" && retainPassword) {
+                if (clone.getName().equals("password") && retainPassword) {
                     clone.setValue(val);
                 } else {
                     clone.setValue(val);
                 }
+                // #645: Allow unresolved i18n keys to pass through for labels and placeholders
+                String label = i18n(request, clone.getLabel());
+                if (!label.startsWith("!")) {
```

Why not change the message source to accept a default and hide this logic away.